### DISCUSSION TITLE
"Invalid Position" errors for several variables reading Arotherm Plus Heat Pump

### DISCUSSION THREAD

**Discussion by Freddiebe** [john30/ebusd#906] [General]
Hi,

Firstly, I would like to say that I am more than happy with my ebus adapter, and all the info it's given me access to. It has enabled me to integrate my heat pump into Home Assistant, and that's fantastic.

I do have a slight problem with a few variables not being reported, that I think is to do with the configuration CSV files being incorrect for my specific hardware / software configuration. My system consists of the following:

![image](https://github.com/john30/ebusd/assets/134110254/12377706-490a-4531-88a1-3f1383faa6c5)

And these are the variables which aren't working:

![image](https://github.com/john30/ebusd/assets/134110254/3ce6c171-0e84-4b03-b856-bab2149a24b6)

My question is, am I using the correct CSV files and are there alternatives which are more suitable? Could I edit the ones in use to make them work with my configuration, and if so, is there a way that I could find the correct parameters for the CSV file?

Many thanks,

Fred

**Comment by Freddiebe** [comment]
Never mind, I should have searched better.

I git cloned the current version and modified the hmu csv as per the following file:

[Link](https://github.com/john30/ebusd-configuration/files/11194651/08.hmu.csv)

I now get many more variables, and most are working with the exception being the yield totals (and a few others).

**Comment by hvbargen** [comment]
I can confirm that this CSV works much better for my Vaillant AroTherm Plus VWL 75/6 A S2.

It took me while how to do use different CSV files, so I'll write a short summary here, as a reference for myself and others (also includes how to use ebusctl).

My configuration is: HomeAssistant runs on Raspi4:
Home Assistant 2023.6.2
Supervisor 2023.06.4
Operating System 10.3
Frontend 20230608.0 - latest

EBUS connector is running on https://adapter.ebusd.eu/v5/ 

Vaillant VWL 75/6 A S2
 VWZ MEH 97/6
VR921
VRC 720 (SensoComfort)
uniSTOR VPS R 100
uniSTOR VIH RW 300

Installation of the adapter was straightforward.
Installation of the ebusd add-on was easy as well.
I already had a working MQTT setup before.

I changed the following config options in the ebusd add-on:
![image](https://github.com/john30/ebusd/assets/37015738/02717e57-0101-4b79-8949-6e577dd521ff)
The IP number is from the adapter. I set up my DHCP server to always use the same IP address for this device.

![image](https://github.com/john30/ebusd/assets/37015738/e5b8b271-1f8d-44a0-a468-b0f3c00245ef)

I disabled the filter, but probably I'll create a more strict filter later, once I know more about the values.

I downloaded a ZIP file containing the config files from https://github.com/john30/ebusd-configuration and unzipped it on my PC.
Then I created another ZIP, containing only the `2.1.x/de` subdirectory.
Using the "File Editor", I uploaded this ZIP into the /config/directory.
Using the SSH&Terminal add-on, I opened a shell on the HA. Inside the shell, I created a new subdirectory ebusd and unzipped the uploaded ZIP file there, so now I have
![image](https://github.com/john30/ebusd/assets/37015738/9894c88a-5680-43c2-9227-6feefc51b3cb)
I downloaded the CSV file from https://github.com/john30/ebusd-configuration/files/11194651/08.hmu.csv to my PC and, 
using the "Upload" button in the "File Editor" again, uploaded it into the vaillant directory on the HA (replacing the original file).
Finally I restarted the add-on.

**Reply by miho9** [reply]
Hi, 

your notes proved really useful to me. Many thanks! I am using a very similar setup. I followed your steps but ended up getting an error message from one of the many other .csv files in the vaillant directory.  Apparently all of them are read when starting ebusd. I presume just 08.hmu.csv and the included files are needed? Or are these relying on other files, which are not included explicitely?
